### PR TITLE
Dcir support with json

### PR DESCRIPTION
**Pull request by svandenb-dev**

### CODE REVIEW

File: pyaedt/edb_core/EDB_Data.py [hunk 1]
```diff
@@ -2932,6 +2935,182 @@ def __init__(self, edbutils, db, cell):
...
+        return self._net
+
+    @net.setter
+    def net(self, value):  # pragma: no cover
+        if isinstance(value, str):
+            self._net = value
+
+    @property
+    def node_type(self):  # pragma: no cover
+        """The type of the node."""
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:2969]:
```suggestion
        """Type of the node."""
```

File: pyaedt/edb_core/EDB_Data.py [hunk 2]
```diff
@@ -2932,6 +2935,182 @@ def __init__(self, edbutils, db, cell):
...
+            dict_out[k[1:]] = v
+        return dict_out
+
+    def _read_json(self, node_dict):  # pragma: no cover
+        for k, v in node_dict.items():
+            self.__setattr__(k, v)
+
+
+class Source(object):
+    """Class for handling Siwave sources"""
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:2999]:
```suggestion
    """Handles Siwave sources."""
```

File: pyaedt/edb_core/EDB_Data.py [hunk 3]
```diff
@@ -2932,6 +2935,182 @@ def __init__(self, edbutils, db, cell):
...
+        return self._positive_node
+
+    @positive_node.setter
+    def positive_node(self, value):  # pragma: no cover
+        if isinstance(value, Node):
+            self._positive_node = value
+
+    @property
+    def negative_node(self):  # pragma: no cover
+        """Negative node of the source e.g. reference."""
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:3055]:
```suggestion
        """Negative node of the source."""
```

File: pyaedt/edb_core/EDB_Data.py [hunk 4]
```diff
@@ -2932,6 +2935,182 @@ def __init__(self, edbutils, db, cell):
...
+
+    @negative_node.setter
+    def negative_node(self, value):  # pragma: no cover
+        if isinstance(value, Node):
+            self._negative_node = value
+            #
+
+    @property
+    def amplitude(self):  # pragma: no cover
+        """Amplitude value for the source. Ampere for current source or Volt for voltage source."""
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:3066]:
```suggestion
        """Amplitude value for the source. Either amperes for current source or volts for voltage source."""
```

File: pyaedt/edb_core/EDB_Data.py [hunk 5]
```diff
@@ -4446,10 +4648,117 @@ def import_json(self, input_file):
...
+        positive_node_net="",
+        negative_node_component="",
+        negative_node_net="",
+    ):
+        """Add a source for the current SimulationConfiguration instance.
+
+        Parameters
+        ----------
+        source_type : SourceType
+            The source type defined.
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:4681]:
```suggestion
            Source type that is defined.
```

File: pyaedt/edb_core/EDB_Data.py [hunk 6]
```diff
@@ -4446,10 +4648,117 @@ def import_json(self, input_file):
...
+
+        name : str
+            Source name.
+
+        amplitude : float
+            Source amplitude value, Volt for voltage source, Amper for current one.
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:4687]:
```suggestion
            Source amplitude value. Either amperes for current source or volts for voltage source.
```

File: pyaedt/edb_core/EDB_Data.py (hunk 5 above)
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:4681]:
```suggestion
            Source type that is defined.
```

File: pyaedt/edb_core/EDB_Data.py [hunk 7]
```diff
@@ -4446,10 +4648,117 @@ def import_json(self, input_file):
...
+        name="",
+        amplitude=1.0,
+        phase=0.0,
+        impedance=1.0,
+        positive_node_component="",
+        positive_node_net="",
+        negative_node_component="",
+        negative_node_net="",
+    ):
+        """Add a source for the current SimulationConfiguration instance.
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:4676]: Either use literal tagging around "SimulationConfiguration" instance or write simply as simulation configuration instance.

File: pyaedt/edb_core/components.py [hunk 8]
```diff
@@ -516,6 +517,77 @@ def get_solder_ball_height(self, cmp):
             return cmp_prop.GetSolderBallProperty().GetHeight()
         return False
 
+    @pyaedt_function_handler()
+    def create_source_on_component(self, sources=None):
+        """Create voltage, current source or resistor on component.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:522]:
```suggestion
        """Create voltage, current source, or resistor on component.
```

File: pyaedt/edb_core/components.py [hunk 9]
```diff
@@ -516,6 +517,77 @@ def get_solder_ball_height(self, cmp):
...
+
+        Parameters
+        ----------
+        sources : list[Source]
+            list of EDB_Data.Source object.
+
+        Returns
+        -------
+        double, bool
+            True when succeeded, False when failed.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:532]:
```suggestion
            ``True`` when successful, ``False`` when failed.
```

File: pyaedt/edb_core/components.py [hunk 10]
```diff
@@ -516,6 +517,77 @@ def get_solder_ball_height(self, cmp):
...
+        """
+
+        if not sources:  # pragma: no cover
+            return False
+        if isinstance(sources, Source):  # pragma: no cover
+            sources = [sources]
+        if isinstance(sources, list):  # pragma: no cover
+            for src in sources:
+                if not isinstance(src, Source):  # pragma: no cover
+                    self._logger.error("List of Source object must be passed as argument")
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:543]:
```suggestion
                    self._logger.error("List of source objects must be passed as an argument.")
```

File: pyaedt/edb_core/components.py [hunk 11]
```diff
@@ -738,12 +810,61 @@ def deactivate_rlc_component(self, component=None, create_circuit_port=False):
...
+            pos_pin_term.SetName(component.refdes)
+            neg_pin_term.SetBoundaryType(self._pedb.edb.Cell.Terminal.BoundaryType.PortBoundary)
+            neg_pin_term.SetIsCircuitPort(True)
+            pos_pin_term.SetReferenceTerminal(neg_pin_term)
+            self._logger.info("Component {} has been replaced by port".format(component.refdes))
+            return True
+
     @pyaedt_function_handler()
     def _create_pin_group_terminal(self, pingroup, isref=False):
         """Creates edb pin group terminal from given edb pin group.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:870]:
```suggestion
        """Creates an edb pin group terminal from a given edb pin group.
```

File: pyaedt/edb_core/EDB_Data.py [hunk 12]
```diff
@@ -2932,6 +2935,182 @@ def __init__(self, edbutils, db, cell):
         self.EdbHandler.layout = cell.GetLayout()
 
 
+class Node(object):
+    """Class handling nodes for Siwave sources."""
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:2939]:
```suggestion
    """Provides for handling nodes for Siwave sources."""
```

File: pyaedt/edb_core/components.py [hunk 13]
```diff
@@ -738,12 +810,61 @@ def deactivate_rlc_component(self, component=None, create_circuit_port=False):
...
-            _cmp = convert_py_list_to_net_list([component.refdes])
-            self._components_methods.AddPortOnRlcComponent(self._active_layout, _cmp)
+            self.add_port_on_rlc_component(component.refdes)
             return True
         else:
             return self.set_component_rlc(component.refdes)
 
+    @pyaedt_function_handler()
+    def add_port_on_rlc_component(self, component=None):
+        """Deactivate rlc component and replace it by circuit port.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:820]:
```suggestion
        """Deactivate RLC component and replace it with a circuit port.
```

File: pyaedt/edb_core/siwave.py [hunk 14]
```diff
@@ -1313,91 +1315,101 @@ def configure_siw_analysis_setup(self, simulation_setup=None):
...
+                                simulation_setup.stop_freq,
+                                simulation_setup.step_freq,
+                            )
+                        )
+                if is_ironpython:
+                    simsetup_info.SweepDataList.Add(sweep)
+                else:
+                    simsetup_info.SweepDataList = convert_py_list_to_net_list([sweep])
+            except Exception as err:
+                self._logger.error("Exception in sweep configuration: {0}".format(err))
```
Comment (PipKat) [inline pyaedt/edb_core/siwave.py:1403]:
```suggestion
                self._logger.error("Exception in sweep configuration: {0}.".format(err))
```

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**